File: DefaultCodeInputOutput/5.java
import java.io.*;
import java.util.*;

public class 5 {

    public static boolean isPalindrome(int x) {
        //Write your code here
    }

    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("{\"error\": \"Invalid input\"}");
            return;
        }

        try {
            String input = args[0];
            if (!input.startsWith("{\"x\":") || !input.endsWith("}")) {
                System.out.println("{\"error\": \"Invalid JSON format\"}");
                return;
            }

            String[] parts = input.substring(1, input.length() - 1).split(":");
            if (parts.length != 2 || !parts[0].trim().equals("\"x\"")) {
                System.out.println("{\"error\": \"Invalid JSON format\"}");
                return;
            }

            int x = Integer.parseInt(parts[1].trim());

            boolean result = isPalindrome(x);
            System.out.println("{\"result\": " + result + "}");

        } catch (NumberFormatException e) {
            System.out.println("{\"error\": \"Value is not an integer\"}");
        }
    }
}

File: DefaultCodeInputOutput/5.php
<?php

function is_palindrome($x) {
    //Write your code here
}

function main() {
    global $argv;

    if (count($argv) != 2) {
        echo json_encode(["error" => "Invalid input"]);
        return;
    }

    $input_json = $argv[1];

    $data = json_decode($input_json, true);
    if (json_last_error() !== JSON_ERROR_NONE) {
        echo json_encode(["error" => "Invalid JSON"]);
        return;
    }

    if (!isset($data['x'])) {
        echo json_encode(["error" => "Missing key 'x'"]);
        return;
    }

    $x = $data['x'];
    if (!is_int($x)) {
        echo json_encode(["error" => "Value is not an integer"]);
        return;
    }

    $result = is_palindrome($x);
    echo json_encode(["result" => $result]);
}

main();

?>
